src/main/java/uk/gov/companieshouse/api/strikeoffobjections/file/FileTransferApiClient.java

```diff
@@ -0,0 +1,125 @@
+        FileTransferApiClientResponse response = new FileTransferApiClientResponse();
+
+        try {
+            T operationResponse = operation.execute();
+            response = responseBuilder.createResponse(operationResponse);
+        } catch (IOException e) {
+            logger.errorContext(e.getMessage(), e);
+            response.setHttpStatus(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return response;
+    }
+
+    /**
+     * Uploads a file to the file-transfer-api
+     * Creates a multipart form request containing the file and sends to
+     * the file-transfer-api. The response from the file-transfer-api contains
+     * the new unique id for the file. This is captured and returned in the FileTransferApiClientResponse.
+     * @param fileToUpload The file to upload
+     * @return FileTransferApiClientResponse containing the file id if successful, and http status
+     */
+    public FileTransferApiClientResponse upload(MultipartFile fileToUpload) {
+        return makeApiCall(
+             () -> getFileTransferOperation(fileToUpload),
+             responseEntity ->  getFileTransferApiClientResponse(responseEntity)
+        );
+    }
+
+    /**
+     * @param fileToUpload
+     * @return
+     * @throws IOException
+     */
+    private ResponseEntity<FileTransferApiResponse> getFileTransferOperation(MultipartFile fileToUpload) throws IOException {
+        HttpHeaders headers = createFileTransferApiHttpHeaders();
+        LinkedMultiValueMap<String, String> fileHeaderMap = createUploadFileHeader(fileToUpload);
+        HttpEntity<byte[]> fileHttpEntity = new HttpEntity<>(fileToUpload.getBytes(), fileHeaderMap);
+        LinkedMultiValueMap<String, Object> body = createUploadBody(fileHttpEntity);
+        HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = new HttpEntity<>(body, headers);
+        return restTemplate.postForEntity(fileTransferApiURL, requestEntity, FileTransferApiResponse.class);
+    }
+
+    /**
+     * FileTransferResponseBuilder - the output from FileTransferOperation is the input into
+     * this FileTransferResponseBuilder
+     * @param responseEntity
+     * @return
+     */
+    private FileTransferApiClientResponse getFileTransferApiClientResponse(ResponseEntity<FileTransferApiResponse> responseEntity) {
+        FileTransferApiClientResponse fileTransferApiClientResponse = new FileTransferApiClientResponse();
+        if (responseEntity != null) {
+            fileTransferApiClientResponse.setHeaders(responseEntity.getHeaders());
+            fileTransferApiClientResponse.setHttpStatus(responseEntity.getStatusCode());
+            FileTransferApiResponse apiResponse = responseEntity.getBody();
+            if (apiResponse != null) {
+                fileTransferApiClientResponse.setFileId(apiResponse.getId());
+            }
+        } else {
+            logger.errorContext(String.format("%s %s", NULL_RESPONSE_MESSAGE,fileTransferApiURL));
```

Comment:
Again, I would ensure the requestId is passed in as the context argument.
As this is 'expected' behavior, I don't thi k we need an errorContext here, infoContext(context, message) would be fine

Reply:
Changed to info context.